libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Debugging/ResourceSourceContext.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Bot.Builder.Dialogs.Debugging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Declarative.Debugging
+{
+    /// <summary>
+    /// Resource-aware specialization of <see cref="SourceContext"/>.
+    /// </summary>
+    internal class ResourceSourceContext : SourceContext
+    {
+        /// <summary>
+        /// Gets a mapping between <see cref="JToken" /> and their respective resource ids. 
+        /// </summary>
+        /// <value>
+        /// A mapping between <see cref="JToken" /> and their respective resource ids.
+        /// </value>
+        internal Dictionary<JToken, string> ResourceIdMap { get; } = new Dictionary<JToken, string>(new JTokenEqualityComparer());
```

Comment:
>ResourceIdMap [](start = 44, length = 13)

I would just name this 'DefaultIdMap' and put in SourceContext.

Reply:
For now I updated the name, we can chat the location in person, would want to brainstorm on pros / cons of each approach.

Reply:
Renamed to DefaultIdMap, left it in the current class after our call.